apps/frontend/src/lib/store/cycle.store.ts

```diff
@@ -0,0 +1,79 @@
+import axios from "axios"
+import { create } from "zustand"
+
+import {
+  CreateItemResponse,
+  CycleItem,
+  CycleItems,
+  CycleItemStoreTypes,
+} from "../@types/Items/Cycle"
+import { BACKEND_URL } from "../constants/urls"
+
+export const useCycleItemStore = create<CycleItemStoreTypes>((set, get) => ({
+  cycleItem: null,
+  cycleItems: [],
+  isLoading: false,
+  isFetched: false,
+  setCycleItem: (cycleItem: CycleItem | null) => set({ cycleItem }),
+  setIsFetched: (isFetched: boolean) => set({ isFetched }),
+  fetchItems: async (session: string) => {
+    let cycleItems_: CycleItem[] = []
+    set({ isLoading: true })
+    try {
+      console.log("Fetching items from API...")
+      const { data } = await axios.get<CycleItems>(
+        `${BACKEND_URL}/api/inbox/`,
+        {
+          headers: { Authorization: `Bearer ${session}` },
+        }
+      )
+      console.log("API response:", data)
+      cycleItems_ = data.response
+      console.log("Setting cycleItems in store:", cycleItems_)
+      set({ cycleItems: cycleItems_, isFetched: true })
+      console.log("Store state after update:", get())
+    } catch (error) {
+      console.error("Error fetching cycle items: ", error)
+    } finally {
```

Comment:
From now on it will be better if we start doing better error handling, like if the user token has expired, in that case api returns 401. So we can redirect them to login page and display a message that they've been logged out.

Need to put this check to most of the api calls

Reply:
Otherwise the user will keep editing and their progress can get lost since the apis kept failing and the user never came to know

Reply:
you are right, lets create a issue for this then.